Fix nightly Clippy warnings

tcharding opened the pull request:
### Description

<!-- Describe the purpose of this PR, what's being adding and/or fixed -->

Checking `bdk` with the nightly toolchain throws a bunch of warnings, ~all are trivial to fix~. Turns out one of them is not, the final patch of this PR may or may not be deemed an improvement, please review and I can drop if required.

Fix warnings emitted by Clippy when using nightly toolchain.

### Notes to the reviewers

<!-- In this section you can include notes directed to the reviewers, like explaining why some parts
of the PR were done in a specific way -->

While bdk does not appear to explicitly rely on nightly features I like to use nightly so I can pass `-Zunstable-options` to clippy. I do this so that Clippy doesn't hide warnings after the first time showing them, instead throwing warnings every check. 

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

tcharding committed:
Use assert!(foo) instead of assert_eq!(foo, true)

It is redundant to pass true/false to `assert_eq!` since `assert!`
already asserts true/false.

This may, however, be controversial if someone thinks that

```
    assert_eq!(foo, false);
```

is more clear than

```
    assert!(!foo);
```

Use `assert!` directly instead of `assert_eq!` with true/false argument.

tcharding committed:
Remove unnecessary clone

Clippy emits:

  warning: using `clone` on type `descriptor::policy::Condition` which
  implements the `Copy` trait

Remove the clone and rely on `Copy`.

tcharding committed:
Use !any() instead of find()...is_none()

Clippy emits:

  warning: called `is_none()` after searching an `Iterator` with `find`

As suggested, use the construct: `!foo.iter().any(...)`

tcharding committed:
Use consistent field ordering

Clippy emits:

  warning: struct constructor field order is inconsistent with struct
  definition field order

As suggested, re-order the fields to be consistent with the struct
definition.

RCasatta commented on src/psbt/mod.rs:
```diff
@@ -24,16 +24,17 @@ impl PsbtUtils for PSBT {
-                Some(in_tx.output[tx.input[input_index].previous_output.vout as usize].clone())
-            } else {
-                None
+        match self.inputs.get(input_index) {
```
I do agree this code looks less readable than before, if the specific lint is not usable maybe `#[allow(clippy::all)]` with a comment for this block?

tcharding replied:
Cool, nice idea, I did not think of that. Will update PR as suggested. Thanks.

tcharding committed:
Clear clippy manual_map warning

The lint `manual_map` is new so we cannot explicitly allow it and
maintain backwards comparability. Instead, allow all lints for
`get_utxo_for` with a comment explaining why.

tcharding committed:
Remove unneeded unit expression

Clippy emits:

  warning: unneeded unit expression

As suggested, remove the unneeded unit expression.

tcharding commented:
Please note, this PR now has an additional commit on the end of it (a new nightly Clippy warning snuck in during rebase on master).

notmandatory reviewed (approved):
Look like good cleanups and as usual good commit docs. ACK e1066e9

tcharding commented:
> Look like good cleanups and as usual good commit docs. ACK [e1066e9](https://github.com/bitcoindevkit/bdk/commit/e1066e955cf6c4fc0a20b4a895f064e70c696dae)

Thanks, appreciate the complement.  My commit log habits were formed in the fires of the linux-staging mailing list :)
